src/colorizer/utils/react_utils.ts

```diff
@@ -40,6 +40,34 @@ export function useDebounce<T>(value: T, delayMs?: number): T {
   return debouncedValue;
 }
 
+/**
+ * Combines useTransition and useDebounce. Debounces value and initiates a state transition once the value has come
+ * to rest.
```

Comment:
When is this used, for example, and in general when would someone use this?

Reply:
For the scatterplot tab, it's used to delay rendering the plot when the dataset, track, frame, or other related props are changing, and to mark the resulting state changes as interruptible. I think there's two usage scenarios, which would be the ideal scenario and the current usage in the scatterplot.

In the ideal scenario, you could pass in a value that's changing quickly (say, a slider input) that causes a component to change a lot of DOM elements. By combining the debounce with a transition, the recalculating of the component's DOM elements would only be triggered when the slider input comes to rest, and the resulting DOM update would be marked as non-critical by `useTransition`. That way, React can interrupt replacing those parts of the page with other, more critical UI updates.

The thing to note is that `useTransition` is only useful after the new components have rendered, when React takes the resulting DOM update and applies it to the webpage. It doesn't allow React to parallelize the actual call to the render function, so any calculations done can still slow down the main thread.

In the CURRENT scenario, this hook is... maybe not as useful as I'd like it to be. The biggest problem is that Plotly's re-render process happens synchronously, blocking the main thread until it's finished. This means that the benefit of `useTransition` (to delay React replacing elements of the page from the DOM update) won't provide benefits for Plotly's render. I'm using it just in case there are *some* benefits that React is giving, though it's likely the transition is not doing much if anything.

@frasercl can you check my logic on this?

Reply:
(Also, I will update this comment before merging.)

Reply:
(Also note, I'm likely going to do away with this hook in the next PR update because I've made some pretty major optimizations to Plotly :') )

Reply:
Just happened to be doing some unrelated reading about `useTransition` today and it... broke my brain a bit. Idk if I can be of much help thinking through this hook until I've put more work into actually understanding what `useTransition` does. For the moment, I trust that this is some combination of close-to-working enough and non-mission-critical enough to be comfortable merging it.